06_puppy/alextmz/syncstore.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"errors"
+	"sync"
+)
+
+type SyncStore struct {
+	sync.Map
+}
+
+func NewSyncStore() *SyncStore {
+	a := SyncStore{}
+	return &a
+}
+
+func (m *SyncStore) CreatePuppy(p *Puppy) error {
+	_, ok := m.Load(p.ID)
+	if !ok {
+		m.Store(p.ID, *p)
+		return nil
+	}
+	return errors.New("puppy ID being created already exists")
+}
+
+func (m *SyncStore) ReadPuppy(id uint16) (*Puppy, error) {
+	r, ok := m.Load(id)
```

Comment:
nit pick , you could combine this to one line :
if r, ok := m.Load(id); ok {
puppy := r.(Puppy)
}

Reply:
Thats OK, I usually don't like to combine statements but it does increase readability in this one